test/jasmine/tests/select_test.js

```diff
@@ -552,6 +552,54 @@ describe('@flaky Test select box and lasso in general:', function() {
         .catch(fail)
         .then(done);
     });
+
+    it('should select the right data with the corresponding select direction', function(done) {
+
+        var gd = createGraphDiv();
+
+        var selectPath = [[170 - 68, 100 - 96], [255 - 68, 160 - 96]];
+
+        Plotly.newPlot(gd, [
+            {x: [1, 1, 1, 1, 1, 2, 2, 2, 3, 3], type: 'histogram'},
+            {x: [1, 2, 3], y: [-1, -2, -3], type: 'bar'}
+        ], {
+            width: 400,
+            height: 300,
+            showlegend: false,
+            dragmode: 'select'
+        })
+            .then(function() {
+
+                // expect(gd.fullLayout.selectdirection).toBe('any');
+
+                resetEvents(gd);
+
+                drag(selectPath);
+
+                selectedPromise.then(function() {
+                    expect(selectedData.points.length).toBe(1);
+                    expect(selectedData.points[0].x).toBe(1.8);
+                    expect(selectedData.points[0].y).toBe(3);
+                });
+
+                // do a drag and check that the selection is correct for the default `selectdirection`
+
+                return Plotly.relayout(gd, {selectdirection: 'h'});
+            })
+            .then(function() {
```

Comment:
Yep, this looks good! Make sure you keep chaining the async events - your `selectedPromise.then...` is not being waited for by the following code so it's not entirely clear what sequence things will happen in. Try something like:
```js
Plotly.newPlot(...)
.then(function() {
    resetEvents(gd);
    drag(selectPath);
    return selectedPromise;
})
.then(function() {
    expect(selectedData.points.length).toBe(1);
    // etc

    return Plotly.relayout(...);
})
```